### ISSUE TITLE
MongoDB views cause db push error

### ISSUE THREAD

**Issue by jamiter** [prisma/prisma#18424]
Prisma version: 4.11.0
Provider: MongoDB

I'm trying to set my MongoDB indexes (for `@unique` for example) using the `db push` command. I've added a MongoDB view to the database which causes this error:

```
Namespace <dbName.viewName> is a view, not a collection
```

Removing the view from the database makes `db push` run successfully, setting the desired indexes.

Reproduction:
- Add a view to your MongoDB database
- Run `prisma db push`

Reported this issue in this ticket earlier: https://github.com/prisma/prisma/issues/678#issuecomment-1479625699

Edit: It might be good to mention that this error also shows up when doing a `prisma db pull --force` when there is a view in the database.

**Comment by jamiter** [comment]
Thanks for your response @janpio, but I'm not trying to add an index to the view, but another model. Adding that index, after removing the view from the DB, is actually working.

To be clear, I'm not trying to migrate or push a view. But there being a view in the DB causes this issue.

**Comment by jamiter** [comment]
The schema doesn't affect the error. The fact that a view is available in the database is enough to cause an issue.

Schema:

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["views"]
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model User {
  id     String @id @default(auto()) @map("_id") @db.ObjectId
  email  String @unique

  @@map("users")
}
```

Doint a `db push` will work as expected. A collection is created an a unique index is added to the email.

To have the error occur, add a view. For example, use a seed script with the following:

```typescript
  await prisma.$runCommandRaw({
    create: 'userView',
    viewOn: 'users',
    pipeline: [
      {
        $addFields: {
          foo: "bar"
        }
      },
    ]
  });
```

Trying to run `db push` after the view has been added will cause the mentioned error. Removing the view (manually) will make `db push` work again.

**Comment by jamiter** [comment]
2 months in and I wonder if this is being looked at.

It's as simple as filtering only the collections and ignoring the views, like in this Stack Overflow issue: https://stackoverflow.com/questions/56013171/mongodb-list-collection-without-showing-views

**Comment by jamiter** [comment]
Looks like a duplicate of https://github.com/prisma/prisma/issues/16179

**Comment by jamiter** [comment]
Also related: https://github.com/prisma/prisma/issues/17006

**Comment by christianledgard** [comment]
Any news on this? Having the same issue over here. @jamiter, could you let me know if you found a workaround?

**Comment by jamiter** [comment]
@christianledgard, we currently remove the views before doing the db push. Our seed script ads them again afterwards. This is done automatically in our setup.

In concept it works like this:
- npm run db:remove-views
- npm run db:push
- npm run db:seed // ads views again

Because the views do not contain any data directly this is not an issue. The big downside (besides being a workaround) is that the views do not exist for roughly 10 seconds, which could be an issue on production. And because we want to always be able to release, I hope this will be fixed soon.

It doesn't seem like a lot of work to fix... I tried it myself but didn't get the engines build correctly with rust on my M1.

**Comment by Jolg42** [comment]
Note: This is now fixed in `5.3.0-dev.9` if you want to try it out. Let us know if it works like expected.
It will be released in the official 5.3.0 version on Tuesday, September, 12th.
